synapse/lib/logging.py

```diff
@@ -0,0 +1,265 @@
+import os
+import sys
+import asyncio
+import logging
+import weakref
+import collections
+
+import synapse.exc as s_exc
+import synapse.common as s_common
+
+import synapse.lib.coro as s_coro
+import synapse.lib.json as s_json
+import synapse.lib.const as s_const
+import synapse.lib.scope as s_scope
+
+logger = logging.getLogger(__name__)
+
+logtodo = []
+logbase = None
+loglock = asyncio.Lock()
```

Comment:
These structures ( `loglock` / `lockevnt` )will bind to the currently running loop when they are first locked / waited on; meaning we'll end up with isolation issues if we move to isolated ioloops for testing.

Reply:
Good to know, but IMO not worth worrying about yet. There are several other places where we already have this issue ( lmdb commits etc )